### ISSUE TITLE
bug when searching backwards for word

### ISSUE THREAD

**Issue by kiniggit** [sevdokimov/log-viewer#46]
This error occurs when searching backwards for a given term. It's deterministic, it always happens when reaching the line starting with "2021-03-25 07:02:04.456  INFO 123695 " in the image below. The problem appears to be caused due to the overlapping time range of the new retrieved records with the current ones.

![image](https://user-images.githubusercontent.com/2644049/115127985-942c4100-9f8f-11eb-8f7b-94dcb20200ca.png)

The assertion below is triggered.

```
// file: record-renderer.service.ts
if (end < m.length) {
    SlUtils.assert(Record.compareTo(m[end - 1], m[end]) < 0);
}
```

**Comment by sevdokimov** [comment]
Thank you for the bug report. I don't understand how is it possible. If you search bak from "2021-03-25 07:02:04.456", the backend must return records older than "2021-03-25 07:02:04.456", but it returns "2021-03-25 07:07:33.510". 
Could you attach or send me the log sample to reproduce? (sergey.evdokimov85@gmail.com)

**Comment by kiniggit** [comment]
Yeah, it looks like an off-by-one kind of problem. I've sent you the log file. I've also changed the issue assignee to you, ok?

**Comment by kiniggit** [comment]
Nice catch there. It may be due to the log level, just a guess. The record that is out of place is INFO.

**Comment by kiniggit** [comment]
Check this out: https://stackoverflow.com/questions/586824/why-are-my-log4j-log-file-timestamps-out-of-order-and-how-do-i-fix-it in fact, there are multiple threads writing to same log file.

**Comment by sevdokimov** [comment]
It's sad... I have to make log-viewer tolerant to record disorder.

**Comment by kiniggit** [comment]
Yeah... I've actually seen that before, but I've never realized how common it is. I went through some of my log files in production and I saw several cases of that.

**Comment by sevdokimov** [comment]
LogViewer 0.1.7 works correctly with logs containing disordered messages, but if you have one opened log only.
